packages/contracts/contracts/core/permission/PermissionManager.sol

```diff
@@ -0,0 +1,352 @@
+    /// @param _where The address of the target contract for which `who` recieves permission.
+    /// @param _who The address (EOA or contract) owning the permission.
+    /// @param _permissionID The permission identifier.
+    function _grant(
+        address _where,
+        address _who,
+        bytes32 _permissionID
+    ) internal {
+        _grantWithOracle(_where, _who, _permissionID, IPermissionOracle(ALLOW_FLAG));
+    }
+
+    /// @notice This method is used in the internal `_grant` method of the permission manager.
+    /// @param _where The address of the target contract for which `who` recieves permission.
+    /// @param _who The address (EOA or contract) owning the permission.
+    /// @param _permissionID The permission identifier.
+    /// @param _oracle The PermissionOracle to be used or it is just the ALLOW_FLAG.
+    function _grantWithOracle(
+        address _where,
+        address _who,
+        bytes32 _permissionID,
+        IPermissionOracle _oracle
+    ) internal {
+        if (isImmutable(_where, _permissionID))
+            revert PermissionImmutable({where: _where, permissionID: _permissionID});
+
+        bytes32 permission = permissionHash(_where, _who, _permissionID);
+
+        if (permissions[permission] != UNSET_FLAG) {
+            revert PermissionAlreadyGranted({
+                where: _where,
+                who: _who,
+                permissionID: _permissionID
+            });
+        }
+        permissions[permission] = address(_oracle);
+
+        emit Granted(_permissionID, msg.sender, _who, _where, _oracle);
+    }
+
+    /// @notice This method is used in the public `revoke` method of the permission manager.
+    /// @param _where The address of the target contract for which `who` recieves permission.
+    /// @param _who The address (EOA or contract) owning the permission.
+    /// @param _permissionID The permission identifier.
+    function _revoke(
+        address _where,
+        address _who,
+        bytes32 _permissionID
+    ) internal {
+        if (isImmutable(_where, _permissionID)) {
+            revert PermissionImmutable({where: _where, permissionID: _permissionID});
+        }
+
+        bytes32 permission = permissionHash(_where, _who, _permissionID);
+        if (permissions[permission] == UNSET_FLAG) {
+            revert PermissionAlreadyRevoked({
+                where: _where,
+                who: _who,
+                permissionID: _permissionID
+            });
+        }
+        permissions[permission] = UNSET_FLAG;
+
+        emit Revoked(_permissionID, msg.sender, _who, _where);
+    }
+
+    /// @notice This method is used in the public `makeImmutable` method of the permission manager.
+    /// @param _where The address of the target contract for which the permission is immutable.
+    /// @param _permissionID The permission identifier.
+    function _makeImmutable(address _where, bytes32 _permissionID) internal {
```

Comment:
IMO the word `freeze` was more clear

Reply:
`freeze` makes you assume that there is an `unfreeze` / `thaw` function. What happens in fact is that the variable becomes immutable, which was the reasoning behind the nameing.